### DISCUSSION TITLE
Idiomas na API em Node e no ReactJS

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1587] [Perguntas e Respostas]
Galera, bom dia!

O app que estou fazendo pretendo colocar em dois idiomas apenas, português e inglês.

Como fazem isto no node? (neste ponto apenas para as respostas da api como "User not found..." ou "Usuário não encontrado"), estou fazendo um json com .pt.json e .en.json, onde incluo em meu script e ambos contêm as constantes nos respetivos idiomas, mas não sei se é a melhor escolha...

E como fazer isto no front ReactJS e no App em react-native? Como vocês fazem?

Muito obrigado, abraços!

**Comment by bizarro** [comment]
Com React geralmente nos projetos que eu trabalhei a equipe decidiu usar o [react-intl](https://github.com/formatjs/react-intl) e se não me engano ele é o mais utilizado mesmo. Aí a integração é bem mais fácil, a documentação deles é bem completa e ensina como fazer o processo.

**Comment by ninetails** [comment]
Dá pra passar o idioma do _client_ pro _server_ via [header](https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Headers/Content-Language).

Já pro i18n, não cheguei a usar mas sempre tive vontade sobre o [i18n Tagged Template Literals](http://i18n-tag.kolmer.net/).
